Network/HTTP/Req.hs

```diff
@@ -866,17 +869,31 @@ instance HttpMethod method => RequestComponent (Womb "method" method) where
 --
 -- > https "юникод.рф"
 -- > -- https://%D1%8E%D0%BD%D0%B8%D0%BA%D0%BE%D0%B4.%D1%80%D1%84
-data Url (scheme :: Scheme) = Url Scheme (NonEmpty Text)
-  -- NOTE The second value is the path segments in reversed order.
-  deriving (Eq, Ord, Show, Data, Typeable, Generic)
+data Url (scheme :: Scheme) = Typeable scheme => Url' (NonEmpty Text)
```

Comment:
Yet another option would be to replace the `Typeable scheme` constraint by an `TypeRep scheme` field (from Type.Reflection), which would carry the same information. That way we would get the `Generic` instance, but we'd loose the `Data` instance, because there is no `Data` instance for `TypeRep`. This also has the advantage that we don't need those standalone derivings and can just use normal derive again.

A last option would be to replace it by a `Dict (Typeable scheme)` field with `Dict` from the constraint package. That way we would get both a `Generic` and a `Data` instance without defining any of them, but it would obviously incur a new dependency.

Reply:
It's kind of annoying that `TypeRep` doesn't has a `Data` instance.